src/plugins/condition/ConditionManager.js

```diff
@@ -31,13 +31,41 @@ export default class ConditionManager extends EventEmitter {
         this.conditionSetDomainObject = conditionSetDomainObject;
         this.timeAPI = this.openmct.time;
         this.latestTimestamp = {};
+        this.composition = this.openmct.composition.get(conditionSetDomainObject);
+        this.composition.on('add', this.addTelemetry, this);
```

Comment:
Looks like you're subscribing to any telemetry that's added to the conditionSet. But we should only be subscribing to telemetry that's being used by conditions.

Reply:
Correct. This was done intentionally to reduce calculations per discussion with @deeptailor. There shouldn't be too many instances of telemetry being added and not used. If this needs rework then we should open a new issue and discuss as a team.